docs/okeps/okep-4380-network-qos.md

```diff
@@ -0,0 +1,408 @@
+
+For example, using LGW mode and assuming there's a single node `node1` and the following `NetworkQoS` is created:
+
+```yaml
+kind: NetworkQoS
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: qos-external-foo
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      priority: Critical
+  egress:
+  - priority: 100
+    dscp: 59
+    classifier:
+      ports:
+        protocol: "tcp"
+        port: 22
+      to:
+      - cidr: 1.2.3.4/32
+  - priority: 10
+    dscp: 46
+    classifier:
+      to:
+      - cidr: 1.2.3.4/32
+---
+kind: NetworkQoS
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: qos-external-bar
+  namespace: default
+spec:
+  egress:
+  - dscp: 30
+    classifier:
+      to:
+      - cidr: 0.0.0.0/0
+```
+the equivalent of:
+```bash
+ovn-nbctl qos-add node1 to-lport 1000 "ip4.src == <default_ns-qos-external-foo address set> && ip4.dst == 1.2.3.4/32 && tcp && tcp.dst == 22" dscp=59
+ovn-nbctl qos-add node1 to-lport 999 "ip4.src == <default_ns-qos-external-foo address set> && ip4.dst == 1.2.3.4/32" dscp=46
+ovn-nbctl qos-add node1 to-lport 998 "ip4.src == <default_ns address set> && ip4.dst == 0.0.0.0/0" dscp=30
+```
+will be executed.
+The podSelector in `qos-external-foo` NetworkQoS object is applied for both first and second NetworkQoS rules. So creating a new Pod that matches this selector in the namespace results in its IPs being added to corresponding Address Set.
+The `qos-external-bar` doesn't have podSelector, so NetworkQoS rule specified in this object is applied to all pods in the `default` namespace.
+
+```yaml
+kind: NetworkQoS
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: qos-storage-traffic
+  namespace: games
+spec:
+  podSelector:
+    matchLabels:
+      app: storage-client
+  egress:
+  - priority: 40
+    dscp: 59
+    bandwidth:
+      burst: 100
+      rate: 20000
+    classifier:
+      ports:
+        protocol: "tcp"
+        port: 471
+      to:
+      - podSelector:
+          matchLabels:
+            app: storage-target
+          namespaceSelector:
+            matchLabels:
+            name: storage-target
+---
+kind: NetworkQoS
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: qos-stream-traffic
+  namespace: games
+spec:
+  podSelector:
+    matchLabels:
+      app: stream-client
+  egress:
+  - priority: 40
+    dscp: 33
+    classifier:
+      ports:
+        protocol: "udp"
+        port: 399
+      to:
+      - podSelector:
+          matchLabels:
+          app: stream-client
+        namespaceSelector:
+          matchLabels:
+            name: stream-target
+```
+In the above example, The tcp traffic with dst port 471 from pods with label `app=storage-client` in `games` namespace will have dscp set to 59 and bandwidth limited to specified burst/rate towards pods having label `app=storage-target` residing on the namespace with label `name=storage-target`.
```

Comment:
Could you please add to one of the examples, or on another example a NetworkQos that uses the newly added `networkAttachmentName`. For example:

```yaml
kind: NetworkQoS
apiVersion: k8s.ovn.org/v1
metadata:
  name: qos-external-bar
  namespace: default
spec:
  networkAttachmentName: default/ovn-stream
  egress:
  - dscp: 30
    classifier:
      to:
      - cidr: 0.0.0.0/0
```

Reply:
ok, added this example, please take a look.